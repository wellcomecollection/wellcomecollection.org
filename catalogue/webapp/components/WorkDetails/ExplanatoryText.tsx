import { AppContext } from '@weco/common/views/components/AppContext/AppContext';
import {
  useState,
  useEffect,
  useContext,
  FunctionComponent,
  ReactElement,
  PropsWithChildren,
} from 'react';
import styled from 'styled-components';
import { plus } from '@weco/common/icons';
import { font } from '@weco/common/utils/classnames';
import Icon from '@weco/common/views/components/Icon/Icon';
import Space from '@weco/common/views/components/styled/Space';

type IconContainerProps = {
  open: boolean;
};

const IconContainer = styled.div<IconContainerProps>`
  width: 1em;
  height: 1em;
  border-radius: 50%;
  background: ${props =>
    props.theme.color(props.open ? 'black' : 'accent.green')};

  .icon {
    transition: transform 300ms ease;
  }
`;

const Control = styled.button.attrs({
  className: `plain-button ${font('intb', 5)}`,
})`
  display: flex;
  align-items: center;

  cursor: pointer;
  padding: 0;

  &:focus-visible,
  &:focus {
    outline: 'initial';
  }

  :focus:not(:focus-visible) {
    outline: none;
  }
`;

const ControlText = styled.span`
  text-decoration: underline;
`;

type ContentProps = {
  hidden: boolean;
};

const Content = styled.div<ContentProps>`
  display: ${props => (props.hidden ? 'none' : 'block')};
`;

type Props = PropsWithChildren<{
  id: string;
  controlText: string;
}>;

const ExplanatoryText: FunctionComponent<Props> = ({
  id,
  controlText,
  children,
}) => {
  const { isEnhanced } = useContext(AppContext);
  const [showContent, setShowContent] = useState(true);
  useEffect(() => {
    setShowContent(false);
  }, []);
  return (
    <div>
      {isEnhanced && (
        <Control
          aria-controls={id}
          aria-expanded={showContent}
          onClick={() => {
            setShowContent(!showContent);
          }}
        >
          <Space as="span" h={{ size: 's', properties: ['margin-right'] }}>
            <IconContainer open={showContent}>
              <Icon
                icon={plus}
                rotate={showContent ? 45 : undefined}
                matchText={true}
                iconColor="white"
              />
            </IconContainer>
          </Space>
          <ControlText>{controlText}</ControlText>
        </Control>
      )}
      <Content id={id} aria-hidden={!showContent} hidden={!showContent}>
        <Space
          v={{
            size: 'l',
            properties: ['margin-top'],
          }}
        >
          {children}
        </Space>
      </Content>
    </div>
  );
};

export default ExplanatoryText;
