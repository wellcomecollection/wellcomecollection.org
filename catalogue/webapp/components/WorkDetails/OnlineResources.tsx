import { useEffect, useState, useRef, FunctionComponent } from 'react';
import { font } from '@weco/common/utils/classnames';
import { getItemsByLocationType } from '../../utils/works';
import WorkDetailsSection from '../WorkDetailsSection/WorkDetailsSection';
import { DigitalLocation, Work } from '@weco/common/model/catalogue';
import styled from 'styled-components';

const ShowHideButton = styled.button.attrs({
  className: `plain-button no-margin no-padding ${font('intr', 5)}`,
})`
  text-decoration: underline;

  &:hover {
    text-decoration: none;
  }
`;
type Props = {
  work: Work;
};
const OnlineResources: FunctionComponent<Props> = ({ work }: Props) => {
  const onlineResources = getItemsByLocationType(work, 'online-resource').map(
    i => {
      const [location] = i.locations;
      return {
        title: i.title,
        location,
      };
    }
  ) as { title: string; location: DigitalLocation }[];

  const firstThreeOnlineResources = onlineResources.slice(0, 3);
  const remainingOnlineResources = onlineResources.slice(3);
  const [
    isShowingRemainingOnlineResources,
    setIsShowingRemainingOnlineResources,
  ] = useState(false);
  const firstOfRemainingOnlineResourcesRef = useRef<HTMLAnchorElement>(null);
  const showHideResourcesRef = useRef<HTMLButtonElement>(null);
  const firstRender = useRef(true);
  useEffect(() => {
    if (isShowingRemainingOnlineResources) {
      firstOfRemainingOnlineResourcesRef?.current?.focus();
    } else {
      if (!firstRender.current) {
        showHideResourcesRef?.current?.focus();
      } else {
        firstRender.current = false;
      }
    }
  }, [isShowingRemainingOnlineResources]);

  return onlineResources.length > 0 ? (
    <WorkDetailsSection headingText="Online resources">
      <ul className="plain-list no-margin no-padding">
        {firstThreeOnlineResources.map(item => (
          <li className={font('intr', 5)} key={item.location.url}>
            {item.title && `${item.title}: `}
            <a href={item.location.url}>
              {item.title ? 'View resource' : item.location.linkText}
            </a>
          </li>
        ))}
        {isShowingRemainingOnlineResources && (
          <>
            {remainingOnlineResources.map((item, index) => (
              <li className={font('intr', 5)} key={item.location.url}>
                {item.title && `${item.title}: `}
                <a
                  href={item.location.url}
                  ref={
                    index === 0 ? firstOfRemainingOnlineResourcesRef : undefined
                  }
                >
                  {item.title ? 'View resource' : item.location.linkText}
                </a>
              </li>
            ))}
          </>
        )}
      </ul>
      {remainingOnlineResources.length > 0 && (
        <ShowHideButton
          ref={showHideResourcesRef}
          onClick={() =>
            setIsShowingRemainingOnlineResources(
              !isShowingRemainingOnlineResources
            )
          }
        >
          {isShowingRemainingOnlineResources
            ? '- Show less'
            : `+ ${remainingOnlineResources.length} more`}
        </ShowHideButton>
      )}
    </WorkDetailsSection>
  ) : null;
};

export default OnlineResources;
