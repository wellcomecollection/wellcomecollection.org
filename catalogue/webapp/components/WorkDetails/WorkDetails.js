// @flow
import type { Node } from 'react';
import type { LicenseData } from '@weco/common/utils/get-license-info';
import type { LicenseType } from '@weco/common/model/license';
import { font, spacing, grid, classNames } from '@weco/common/utils/classnames';
import { worksUrl } from '@weco/common/services/catalogue/urls';
import { Fragment } from 'react';
import SpacingComponent from '@weco/common/views/components/SpacingComponent/SpacingComponent';
import SpacingSection from '@weco/common/views/components/SpacingSection/SpacingSection';
import Icon from '@weco/common/views/components/Icon/Icon';
import Divider from '@weco/common/views/components/Divider/Divider';
import CopyUrl from '@weco/common/views/components/CopyUrl/CopyUrl';
import MetaUnit from '@weco/common/views/components/MetaUnit/MetaUnit';
import Layout12 from '@weco/common/views/components/Layout12/Layout12';
import Download from '../Download/Download';

type WorkDetailsSectionProps = {|
  headingText?: string,
  children: Node,
|};

const WorkDetailsSection = ({
  headingText,
  children,
}: WorkDetailsSectionProps) => {
  return (
    <div
      className={classNames({
        grid: true,
      })}
    >
      <div
        className={classNames({
          [grid({ s: 12, m: 12, l: 4, xl: 4 })]: true,
        })}
      >
        {headingText && (
          <h2
            className={classNames({
              [font({ s: 'WB6', m: 'WB5' })]: true,
              'work-details-heading': true,
            })}
          >
            {headingText}
          </h2>
        )}
      </div>

      <div
        className={classNames({
          [grid({ s: 12, m: 12, l: 8, xl: 7 })]: true,
        })}
      >
        {children}
      </div>
    </div>
  );
};

type Work = Object;

type Props = {|
  work: Work,
  iiifImageLocationUrl: ?string,
  licenseInfo: ?LicenseData,
  iiifImageLocationCredit: ?string,
  iiifImageLocationLicenseId: ?LicenseType,
  encoreLink: ?string,
|};

const WorkDetails = ({
  work,
  iiifImageLocationUrl,
  licenseInfo,
  iiifImageLocationCredit,
  iiifImageLocationLicenseId,
  encoreLink,
}: Props) => {
  const singularWorkTypeLabel = work.workType.label
    ? work.workType.label.replace(/s$/g, '').toLowerCase()
    : 'item';
  const isbnIdentifiers = work.identifiers.filter(id => {
    return id.identifierType.id === 'isbn';
  });

  const WorkDetailsSections = [];
  if (iiifImageLocationUrl) {
    WorkDetailsSections.push(
      <WorkDetailsSection>
        <Download
          work={work}
          iiifImageLocationUrl={iiifImageLocationUrl}
          licenseInfo={licenseInfo}
          iiifImageLocationCredit={iiifImageLocationCredit}
          iiifImageLocationLicenseId={iiifImageLocationLicenseId}
        />
      </WorkDetailsSection>
    );
  }
  if (
    work.description ||
    work.production.length > 0 ||
    work.physicalDescription ||
    work.extent ||
    work.dimensions ||
    work.lettering ||
    work.genres.length > 0 ||
    work.language
  ) {
    WorkDetailsSections.push(
      <WorkDetailsSection headingText={`About this ${singularWorkTypeLabel}`}>
        {work.description && (
          <MetaUnit
            headingLevel={3}
            headingText="Description"
            text={[work.description]}
          />
        )}

        {work.contributors.length > 0 && (
          <MetaUnit
            headingLevel={3}
            headingText="Contributors"
            text={[
              work.contributors
                .map(contributor => contributor.agent.label)
                .join(' | '),
            ]}
          />
        )}

        {work.production.length > 0 && (
          <MetaUnit
            headingLevel={3}
            headingText="Publication/Creation"
            text={work.production.map(productionEvent => productionEvent.label)}
          />
        )}

        {(work.physicalDescription || work.extent || work.dimensions) && (
          <MetaUnit
            headingLevel={3}
            headingText="Physical description"
            text={[
              [work.extent, work.physicalDescription, work.dimensions]
                .filter(Boolean)
                .join(' '),
            ]}
          />
        )}

        {work.lettering && (
          <MetaUnit
            headingLevel={3}
            headingText="Lettering"
            text={[work.lettering]}
          />
        )}

        {work.genres.length > 0 && (
          <MetaUnit
            headingLevel={3}
            headingText="Type/Technique"
            tags={work.genres.map(g => {
              return {
                textParts: g.concepts.map(c => c.label),
                linkAttributes: worksUrl({
                  query: `"${g.label}"`,
                  page: 1,
                }),
              };
            })}
          />
        )}

        {work.language && (
          <MetaUnit
            headingLevel={3}
            headingText="Language"
            links={[work.language.label]}
          />
        )}
      </WorkDetailsSection>
    );
  }
  if (work.subjects.length > 0) {
    WorkDetailsSections.push(
      <WorkDetailsSection headingText="Subjects">
        <MetaUnit
          tags={work.subjects.map(s => {
            return {
              textParts: s.concepts.map(c => c.label),
              linkAttributes: worksUrl({
                query: `"${s.label}"`,
                page: 1,
              }),
            };
          })}
        />
      </WorkDetailsSection>
    );
  }
  if (encoreLink) {
    WorkDetailsSections.push(
      <WorkDetailsSection headingText="Find in the library">
        <MetaUnit
          text={[
            `This ${singularWorkTypeLabel} is available at <a href="${encoreLink}">Wellcome library</a>`,
          ]}
        />
      </WorkDetailsSection>
    );
  }
  WorkDetailsSections.push(
    <WorkDetailsSection headingText="Identifiers">
      {isbnIdentifiers.length > 0 && (
        <MetaUnit
          headingText="ISBN"
          list={isbnIdentifiers.map(id => id.value)}
        />
      )}
      <MetaUnit headingText="Share">
        <CopyUrl
          id={work.id}
          url={`https://wellcomecollection.org/works/${work.id}`}
        />
      </MetaUnit>
    </WorkDetailsSection>
  );
  if (licenseInfo) {
    WorkDetailsSections.push(
      <WorkDetailsSection headingText="License information">
        <div id="licenseInformation">
          <MetaUnit
            headingLevel={3}
            headingText="License information"
            text={licenseInfo.humanReadableText}
          />
          <MetaUnit
            headingLevel={3}
            headingText="Credit"
            text={[
              `${work.title.replace(/\.$/g, '')}.${' '}
              ${
                iiifImageLocationCredit
                  ? `Credit: <a href="https://wellcomecollection.org/works/${
                      work.id
                    }">${iiifImageLocationCredit}</a>. `
                  : ` `
              }
              ${
                licenseInfo.url
                  ? `<a href="${licenseInfo.url}">${licenseInfo.text}</a>`
                  : licenseInfo.text
              }`,
            ]}
          />
        </div>
      </WorkDetailsSection>
    );
  }
  WorkDetailsSections.push(
    <WorkDetailsSection>
      <div className="flex flex--v-center">
        <Icon name="underConstruction" extraClasses="margin-right-s2" />
        <p
          className={`${font({
            s: 'HNL5',
            m: 'HNL4',
          })} no-margin`}
        >
          Weâ€™re improving the information on this page.{' '}
          <a href="/works/progress">Find out more</a>.
        </p>
      </div>
    </WorkDetailsSection>
  );

  return (
    <div
      className={classNames({
        row: true,
        'bg-cream': true,
        [spacing({ s: 6, m: 8 }, { padding: ['top', 'bottom'] })]: true,
      })}
    >
      <Layout12>
        {WorkDetailsSections.map((section, i) => {
          return (
            <Fragment key={i}>
              {i > 0 && (
                <>
                  <Divider extraClasses="divider--pumice divider--keyline" />
                  <SpacingComponent />
                </>
              )}
              <SpacingSection>{section}</SpacingSection>
            </Fragment>
          );
        })}
      </Layout12>
    </div>
  );
};

export default WorkDetails;
