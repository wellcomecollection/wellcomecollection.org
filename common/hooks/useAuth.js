// @flow
import Router from 'next/router';
import { useEffect, useState } from 'react';
import fetch from 'isomorphic-unfetch';
import uuidv4 from 'uuid/v4';
import crypto from 'crypto-browserify';
import base64url from 'base64url';

const authDomain = 'https://id.wellcomecollection.org';
const authParams = {
  response_type: 'code',
  // client_id: '5n4vt54rjsg6t691c5b5kiacdv', // localhost
  client_id: '4sl9v9v3i72fs66i0kpgqent8b', // production
  scope: ['openid'].join(' '),
};

function getFromLocalStorage(key: string) {
  return window.localStorage.getItem(key);
}

function getLocalStorageJson(key: string) {
  try {
    return JSON.parse(getFromLocalStorage(key));
  } catch (e) {
    return null;
  }
}

function getRedirectUrl() {
  return `${window.location.href.split('/')[0]}//${
    window.location.href.split('/')[2]
  }/works/auth-code`;
}

async function getToken(code: string, verifier: string) {
  const url = `${authDomain}/oauth2/token`;
  const params = {
    grant_type: 'authorization_code',
    client_id: authParams.client_id,
    redirect_uri: getRedirectUrl(),
    code_verifier: verifier,
    code: code,
  };
  const token = await fetch(url, {
    method: 'POST',
    headers: {
      'Content-Type': 'application/x-www-form-urlencoded',
    },
    body: new URLSearchParams(params),
  }).then(r => {
    if (r.status === 200) return r.json();
    console.error('invalid auth.code');
  });

  return token;
}

function createLoginUrlWithVerifier() {
  const verifier = base64url(uuidv4());
  const challenge = base64url(
    crypto
      .createHash('sha256')
      .update(verifier)
      .digest()
  );

  const loginUrl = `${authDomain}/oauth2/authorize?${new URLSearchParams({
    ...authParams,
    redirect_uri: getRedirectUrl(),
    code_challenge: challenge,
    code_challenge_method: 'S256',
  }).toString()}`;

  return { loginUrl, verifier };
}

type Token = {
  id_token: string,
  access_token: string,
  expired_in: number,
  refresh_token: string,
  token_type: string,
};

const authStates = {
  unauthorized: 'unauthorized',
  authorizing: 'authorizing',
  authorized: 'authorized',
  expired: 'expired',
};

export type Uninitialized = {|
  type: 'uninitialized',
|};

export type Unauthorized = {|
  type: 'unauthorized',
  loginUrl: string,
|};

export type Authorizing = {|
  type: 'authorizing',
|};

export type Authorized = {|
  type: 'authorized',
  token: Token,
|};

export type Expired = {|
  type: 'expired',
|};

type State = Uninitialized | Unauthorized | Authorizing | Authorized | Expired;

const useAuth = () => {
  const [state, setState] = useState<State>({ type: 'uninitialized' });

  // Get the inital state
  useEffect(() => {
    const token = getLocalStorageJson('auth.token');
    const code = Router.query.code;

    if (token) {
      setState(
        ({
          type: authStates.authorized,
          token: token,
        }: Authorized)
      );
    } else if (code) {
      setState(({ type: authStates.authorizing }: Authorizing));
      const code = Router.query.code;
      if (code) {
        authorise(code);
      }
    } else {
      const { verifier, loginUrl } = createLoginUrlWithVerifier();
      window.localStorage.setItem('auth.verifier', verifier);
      setState(({ type: authStates.unauthorized, loginUrl }: Unauthorized));
    }
  }, []);

  async function authorise(code: string) {
    const verifier = window.localStorage.getItem('auth.verifier');
    const token = await getToken(code, verifier);

    if (token) {
      window.localStorage.setItem('auth.token', JSON.stringify(token));
      setState({ type: authStates.authorized, token });
      const link = {
        pathname: Router.asPath.split('?')[0],
        // TODO: This is very app specific, but it's an absolute pain to try and get this to work
        // The side effect is that it removes the rest of the URL params if there are any
        query: {
          action: Router.query.action,
        },
      };
      Router.replace(link, link, { shallow: true });
    } else {
      const { verifier, loginUrl } = createLoginUrlWithVerifier();
      window.localStorage.setItem('auth.verifier', verifier);
      setState(({ type: authStates.unauthorized, loginUrl }: Unauthorized));
      // TODO: nicer solution than window.alert
      window.alert('Authorisation failed. Please try to log in again.');
    }
  }

  return state;
};

export default useAuth;
