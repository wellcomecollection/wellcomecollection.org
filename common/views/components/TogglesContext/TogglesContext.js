// @flow
import { createContext } from 'react';
/* eslint-disable */
const TogglesContext = createContext<{ [key: string]: boolean }>({});
/* eslint-enable */
export default TogglesContext;
