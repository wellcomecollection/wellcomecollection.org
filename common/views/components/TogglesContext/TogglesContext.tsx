import { createContext } from 'react';
import { Toggles } from '@weco/toggles';
import cookies from 'next-cookies';
import { NextPageContext } from 'next';
import { toggles as togglesCache } from '../../../services/cache';

export const parseTogglesFromPageContext = (
  context: NextPageContext
): Toggles => {
  const allCookies = cookies(context);
  const cookiedToggles: Toggles = togglesCache.toggles
    .map(toggle => {
      return {
        id: toggle.id,
        active:
          allCookies[`toggle_${toggle.id}`] === 'true' ?? toggle.defaultValue,
      };
    })
    .reduce((acc, toggle) => {
      return {
        ...acc,
        [toggle.id]: toggle.active,
      };
    }, {});

  return cookiedToggles;
};

const TogglesContext = createContext<Toggles>({} as Toggles);
export default TogglesContext;
