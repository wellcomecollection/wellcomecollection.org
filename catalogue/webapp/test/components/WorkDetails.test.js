// @flow

import {
  mountWithTheme,
  updateWrapperAsync,
} from '@weco/common/test/fixtures/enzyme-helpers';
import WorkDetails from '@weco/catalogue/components/WorkDetails/WorkDetails';
import useAuth from '@weco/common/hooks/useAuth';
import mockAuthStates from '@weco/catalogue/__mocks__/auth-states';
import catalogueWork from '@weco/catalogue/__mocks__/catalogue-work';
import mockStacksWork from '@weco/catalogue/__mocks__/stacks-work';
import TogglesContext from '@weco/common/views/components/TogglesContext/TogglesContext';

const requestableCatalogueWork = catalogueWork;
requestableCatalogueWork.items[0].requestable = true;

jest.mock('@weco/catalogue/services/stacks/items', () => {
  return {
    __esModule: true,
    default: () => new Promise(resolve => resolve(mockStacksWork)),
  };
});

jest.mock('next/router', () => ({
  query: {
    code: '',
  },
}));

jest.mock('next/link', () => ({ children }) => children);

jest.mock('@weco/common/hooks/useAuth');

describe('Feature: 2. As a library member I want to request an item', () => {
  test(`
    Scenario: I'm logged out, viewing a work page
      Given the work has a requestable item
      And it’s available to request
      Then I can see a primary CTA to log in
    `, async () => {
    (useAuth: any).mockReturnValue(mockAuthStates.unauthorized);
    const wrapper = mountWithTheme(
      <TogglesContext.Provider value={{ stacksRequestService: true }}>
        <WorkDetails
          work={requestableCatalogueWork}
          iiifPresentationManifest={null}
          itemUrl={null}
          childManifestsCount={1}
          imageCount={1}
        />
      </TogglesContext.Provider>
    );
    await updateWrapperAsync(wrapper);

    expect(wrapper.find("[data-test-id='libraryLoginCTA']")).toMatchSnapshot();
  });

  test(`Scenario: I'm logged out, viewing a work page
      Given the work has a requestable item
      And it’s unavailable to request
      Then I can't see a primary CTA to log in`, async () => {
    (useAuth: any).mockReturnValue(mockAuthStates.unauthorized);
    const wrapper = mountWithTheme(
      <WorkDetails
        work={catalogueWork}
        iiifPresentationManifest={null}
        itemUrl={null}
        childManifestsCount={1}
        imageCount={1}
      />
    );
    await updateWrapperAsync(wrapper);
    expect(wrapper.find("[data-test-id='libraryLoginCTA']")).toMatchSnapshot();
  });

  // we currently make no distinction between unrequestable and unavailable
  (test: any).todo(`Scenario: I'm logged out, viewing a work page
    Given the work has no requestable items
    Then I can't see a primary CTA to log in`);

  test(`Scenario: I'm logged in, viewing a work page
  Given the work has a requestable item
  And it’s available to request
  Then I can see a primary CTA to request it`, async () => {
    (useAuth: any).mockImplementation(() => {
      return mockAuthStates.authorized;
    });
    const wrapper = mountWithTheme(
      <TogglesContext.Provider value={{ stacksRequestService: true }}>
        <WorkDetails
          work={requestableCatalogueWork}
          iiifPresentationManifest={null}
          itemUrl={null}
          childManifestsCount={1}
          imageCount={1}
        />
      </TogglesContext.Provider>
    );
    await updateWrapperAsync(wrapper);

    expect(wrapper.find("[data-test-id='requestModalCTA']")).toMatchSnapshot();
  });

  (test: any).todo(`Scenario: A work has a single requestable item
  Given that I am on a work page that has a single requestable item
  When a user clicks the primary CTA to request the item
  Then I receive feedback as to whether the request succeeded or failed
  And if successful, I can see information about where to pick up the item`);

  (test: any).todo(`Scenario: A work has multiple requestable items
  Given that I am on a work page that has multiple requestable items
  Then I am able to select which items I want to request
  And click the primary CTA to request the items
  Then I receive feedback as to which item request succeeded or failed
  And if successful, I can see information about where to pick up the items`);
});
