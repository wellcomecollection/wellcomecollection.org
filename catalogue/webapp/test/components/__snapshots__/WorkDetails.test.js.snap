// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feature: 2. As a library member I want to request an item 
    Scenario: I'm logged out, viewing a work page
      Given the work has a requestable item
      And it’s available to request
      Then I can see a primary CTA to log in
     1`] = `
<div
  data-test-id="libraryLoginCTA"
>
  <ButtonSolidLink
    clickHandler={[Function]}
    link="/loginUrl"
    text="Log in to request"
    trackingEvent={
      Object {
        "action": "follow Cognito login link",
        "category": "Button",
        "label": "{workId}",
      }
    }
  >
    <ConditionalWrapper
      condition={false}
      wrapper={[Function]}
    >
      <ButtonSolid__SolidButton
        href="/loginUrl"
        onClick={[Function]}
      >
        <StyledComponent
          forwardedComponent={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [
                [Function],
                [Function],
              ],
              "componentStyle": ComponentStyle {
                "componentId": "ButtonSolid__SolidButton-is2j2c-3",
                "isStatic": false,
                "lastClassName": "bADJUb",
                "rules": Array [
                  "display:inline-flex;line-height:1;border-radius:",
                  [Function],
                  ";text-decoration:none;text-align:center;transition:background ",
                  [Function],
                  ",border-color ",
                  [Function],
                  ";border:0;white-space:nowrap;padding:13px 20px;cursor:pointer;&:focus{outline:0;.is-keyboard &{box-shadow:",
                  [Function],
                  ";}}&[disabled],&.disabled{background:",
                  [Function],
                  ";border-color:",
                  [Function],
                  ";color:",
                  [Function],
                  ";cursor:not-allowed;&:hover{text-decoration:none;}}&.disabled{pointer-events:none;}.icon__shape{transition:fill ",
                  [Function],
                  ";fill:currentColor;}.icon__stroke{transition:stroke ",
                  [Function],
                  ";stroke:currentColor;}",
                  "background:",
                  [Function],
                  ";color:",
                  [Function],
                  ";border:2px solid ",
                  [Function],
                  ";",
                  [Function],
                  " &:not([disabled]):hover{background:",
                  [Function],
                  ";border-color:",
                  [Function],
                  ";}",
                ],
              },
              "displayName": "ButtonSolid__SolidButton",
              "foldedComponentIds": Array [
                "ButtonSolid__BaseButton-is2j2c-0",
              ],
              "render": [Function],
              "styledComponentId": "ButtonSolid__SolidButton-is2j2c-3",
              "target": "button",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            }
          }
          forwardedRef={null}
          href="/loginUrl"
          onClick={[Function]}
        >
          <a
            className="ButtonSolid__BaseButton-is2j2c-0 ButtonSolid__SolidButton-is2j2c-3 bADJUb link-reset"
            href="/loginUrl"
            onClick={[Function]}
          >
            <ButtonSolid__BaseButtonInner>
              <StyledComponent
                forwardedComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [
                      [Function],
                    ],
                    "componentStyle": ComponentStyle {
                      "componentId": "ButtonSolid__BaseButtonInner-is2j2c-1",
                      "isStatic": false,
                      "lastClassName": "dxoWEE",
                      "rules": Array [
                        "height:1em;",
                      ],
                    },
                    "displayName": "ButtonSolid__BaseButtonInner",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "ButtonSolid__BaseButtonInner-is2j2c-1",
                    "target": "span",
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  }
                }
                forwardedRef={null}
              >
                <span
                  className="ButtonSolid__BaseButtonInner-is2j2c-1 dxoWEE font-hnm font-size-5  flex flex--v-center"
                >
                  Log in to request
                </span>
              </StyledComponent>
            </ButtonSolid__BaseButtonInner>
          </a>
        </StyledComponent>
      </ButtonSolid__SolidButton>
    </ConditionalWrapper>
  </ButtonSolidLink>
</div>
`;

exports[`Feature: 2. As a library member I want to request an item Scenario: I'm logged in, viewing a work page
  Given the work has a requestable item
  And it’s available to request
  Then I can see a primary CTA to request it 1`] = `
<div
  data-test-id="requestModalCTA"
>
  <ForwardRef
    clickHandler={[Function]}
    text="Request"
    trackingEvent={
      Object {
        "action": "open Stacks request modal window",
        "category": "Button",
        "label": "{workId}",
      }
    }
  >
    <ButtonSolid__SolidButton
      onClick={[Function]}
    >
      <StyledComponent
        forwardedComponent={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [
              [Function],
              [Function],
            ],
            "componentStyle": ComponentStyle {
              "componentId": "ButtonSolid__SolidButton-is2j2c-3",
              "isStatic": false,
              "lastClassName": "bADJUb",
              "rules": Array [
                "display:inline-flex;line-height:1;border-radius:",
                [Function],
                ";text-decoration:none;text-align:center;transition:background ",
                [Function],
                ",border-color ",
                [Function],
                ";border:0;white-space:nowrap;padding:13px 20px;cursor:pointer;&:focus{outline:0;.is-keyboard &{box-shadow:",
                [Function],
                ";}}&[disabled],&.disabled{background:",
                [Function],
                ";border-color:",
                [Function],
                ";color:",
                [Function],
                ";cursor:not-allowed;&:hover{text-decoration:none;}}&.disabled{pointer-events:none;}.icon__shape{transition:fill ",
                [Function],
                ";fill:currentColor;}.icon__stroke{transition:stroke ",
                [Function],
                ";stroke:currentColor;}",
                "background:",
                [Function],
                ";color:",
                [Function],
                ";border:2px solid ",
                [Function],
                ";",
                [Function],
                " &:not([disabled]):hover{background:",
                [Function],
                ";border-color:",
                [Function],
                ";}",
              ],
            },
            "displayName": "ButtonSolid__SolidButton",
            "foldedComponentIds": Array [
              "ButtonSolid__BaseButton-is2j2c-0",
            ],
            "render": [Function],
            "styledComponentId": "ButtonSolid__SolidButton-is2j2c-3",
            "target": "button",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          }
        }
        forwardedRef={null}
        onClick={[Function]}
      >
        <button
          className="ButtonSolid__BaseButton-is2j2c-0 ButtonSolid__SolidButton-is2j2c-3 bADJUb"
          onClick={[Function]}
        >
          <ButtonSolid__BaseButtonInner>
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [
                    [Function],
                  ],
                  "componentStyle": ComponentStyle {
                    "componentId": "ButtonSolid__BaseButtonInner-is2j2c-1",
                    "isStatic": false,
                    "lastClassName": "dxoWEE",
                    "rules": Array [
                      "height:1em;",
                    ],
                  },
                  "displayName": "ButtonSolid__BaseButtonInner",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "ButtonSolid__BaseButtonInner-is2j2c-1",
                  "target": "span",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
            >
              <span
                className="ButtonSolid__BaseButtonInner-is2j2c-1 dxoWEE font-hnm font-size-5  flex flex--v-center"
              >
                <span
                  className=""
                >
                  Request
                </span>
              </span>
            </StyledComponent>
          </ButtonSolid__BaseButtonInner>
        </button>
      </StyledComponent>
    </ButtonSolid__SolidButton>
  </ForwardRef>
  <Modal
    isActive={false}
    setIsActive={[Function]}
  />
</div>
`;

exports[`Feature: 2. As a library member I want to request an item Scenario: I'm logged out, viewing a work page
      Given the work has a requestable item
      And it’s unavailable to request
      Then I can't see a primary CTA to log in 1`] = `null`;
